solutions/day-19-20-21/level1/src/components/CatCard.js

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+import "../styles/CatCard.css";
+
+const CatCard = ({ cat }) => {
+  const [imageURL, setImageURL] = useState();
+
+  useEffect(() => {
+    const controller = new AbortController();
+
+    axios
+      .get(`https://api.thecatapi.com/v1/images/search?breed_id=${cat.id}`, {
+        signal: controller.signal,
+      })
+      .then((response) => {
+        setImageURL(response.data[0].url);
+      });
+
+    return () => {
+      controller.abort();
+    };
+  }, [cat.id]);
+
+  return (
+    <div className="card">
+      <div className="cat-image">
+        {imageURL && <img src={imageURL} alt="Cat" />}
+      </div>
+      <div className="cat-info">
+        <div className="breed-name">{cat.name.toUpperCase()}</div>
```

Comment:
Always use the optional chaining operator `?.` when using js methods (e.g., `.map`, `.toUpperCase`, etc.) to avoid throwing an error in case the `cat.name` is undefined.

Reply:
Updated. Thanks, @ianabalus.